### PR TITLE
[Gallery] Implement Cupertino tab bar demo

### PR DESCRIPTION
**Pull request by rami-a**
Closes https://github.com/material-components/material-components-flutter-gallery/issues/95

Simulator Screen Shot - iPhone 11 Pro - 2019-12-18 at 11 58 17

### CODE REVIEW

File: gallery/gallery/lib/demos/cupertino/cupertino_tab_bar_demo.dart [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            for (final tabInfo in _tabInfo)
+              BottomNavigationBarItem(
+                title: Text(tabInfo.title),
+                icon: Icon(tabInfo.icon),
+              ),
+          ],
+        ),
+        tabBuilder: (context, index) {
+          switch (index) {
+            case 0:
```
Comment (johnsonmh) [inline gallery/gallery/lib/demos/cupertino/cupertino_tab_bar_demo.dart]: Looks like you might be able to simplify this switch statement with:

```dart
tabBuilder: (context, index) {
  return CupertinoTabView(
    builder: (context) => _CupertinoDemoTab(
      title: _tabInfo[index].title,
      icon: _tabInfo[index].icon,
    ),
    defaultTitle: _tabInfo[index].title,
  );
}
```

Comment (rami-a) [reply gallery/gallery/lib/demos/cupertino/cupertino_tab_bar_demo.dart]: Done

File: gallery/gallery/lib/demos/cupertino/cupertino_tab_bar_demo.dart [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  const _TabInfo(this.title, this.icon);
+
+  final String title;
+  final IconData icon;
+}
+
+class CupertinoTabBarDemo extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final _tabInfo = <_TabInfo>[
```
Comment (RedBrogdon) [inline gallery/gallery/lib/demos/cupertino/cupertino_tab_bar_demo.dart]: The `<TabInfo>` here can be dropped. Dart should infer that type correctly.

File: gallery/gallery/lib/demos/cupertino/cupertino_tab_bar_demo.dart [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    Key key,
+    @required this.title,
+    @required this.icon,
+  }) : super(key: key);
+
+  final String title;
+  final IconData icon;
+
+  @override
+  Widget build(BuildContext context) {
```
Comment (RedBrogdon) [inline gallery/gallery/lib/demos/cupertino/cupertino_tab_bar_demo.dart]: Entirely optional (and should be done later even if you decide to do it), but consider giving these scaffolds the ability to navigate to a popup. One of the interesting differences between Cupertino's bottom nav and Android's is that each CupertinoTabView has its own navigation stack, so you can bring up a popup, go to another tab, then return to the original and see the popup still there.

Comment (rami-a) [reply gallery/gallery/lib/demos/cupertino/cupertino_tab_bar_demo.dart]: That's a good idea, I think the demo in the old gallery did this too.

File: gallery/gallery/lib/demos/cupertino/cupertino_tab_bar_demo.dart (hunk 2 above)
Comment (rami-a) [reply gallery/gallery/lib/demos/cupertino/cupertino_tab_bar_demo.dart]: Done

**Pull request merged**